src/main/java/bio/terra/common/flagsmith/FlagsmithService.java

```diff
@@ -32,25 +38,62 @@ public Optional<Boolean> isFeatureEnabled(String feature) {
       return Optional.empty();
     }
 
+    FlagsmithCacheConfig.Builder cacheConfig =
+        FlagsmithCacheConfig.newBuilder().expireAfterAccess(12, TimeUnit.HOURS);
+    if (StringUtils.isNotEmpty(flagsmithProperties.getEnvCacheKey())) {
+      cacheConfig.enableEnvLevelCaching(flagsmithProperties.getEnvCacheKey());
+    }
     var flagsmith =
         FlagsmithClient.newBuilder()
             .setApiKey(flagsmithProperties.getServerSideApiKey())
             .withApiUrl(flagsmithProperties.getApiUrl())
-            .withCache(FlagsmithCacheConfig.newBuilder().build())
+            .withCache(cacheConfig.build())
             .build();
 
-    Flags flags;
-    try {
-      flags = flagsmith.getEnvironmentFlags();
-    } catch (FlagsmithClientError e) {
-      LOGGER.warn("Failed to get environment flags", e);
-      return Optional.empty();
-    }
     try {
+      Flags flags = getWithRetryOnException(flagsmith::getEnvironmentFlags);
       return Optional.of(flags.isFeatureEnabled(feature));
     } catch (FlagsmithClientError e) {
-      LOGGER.warn("Failed to get the feature state of {}", feature, e);
-      return Optional.empty();
+      LOGGER.warn("Feature {} not found in {}", feature, flagsmithProperties.getApiUrl(), e);
+    } catch (Exception e) {
+      LOGGER.warn("Failed to fetch feature {}", feature, e);
+    }
+    return Optional.empty();
+  }
+
+  private static <T> T getWithRetryOnException(SupplierWithException<T> supplier) throws Exception {
```

Comment:
I should move the RetryUtils into common lib... oh well.

Reply:
yeah that would have been really nice :)